lib/lte_link_control/lte_lc.c

```diff
@@ -493,66 +496,61 @@ static void at_handler(void *context, const char *response)
 static int enable_notifications(void)
 {
 	int err;
-	char buf_sub[35];
 
 	/* +CEREG notifications, level 5 */
-	err = at_cmd_write(cereg_5_subscribe, NULL, 0, NULL);
+	err = nrf_modem_at_printf(AT_CEREG_5);
 	if (err) {
 		LOG_ERR("Failed to subscribe to CEREG notifications");
 		return err;
 	}
 
 	if (IS_ENABLED(CONFIG_LTE_LC_TAU_PRE_WARNING_NOTIFICATIONS)) {
-		snprintk(buf_sub,
-			 sizeof(buf_sub),
-			 AT_XT3412_SUB,
-			 CONFIG_LTE_LC_TAU_PRE_WARNING_TIME_MS,
-			 CONFIG_LTE_LC_TAU_PRE_WARNING_THRESHOLD_MS);
-
-		/* %XT3412 notifications subscribe */
-		err = at_cmd_write(buf_sub, NULL, 0, NULL);
+		err = nrf_modem_at_printf(AT_XT3412_SUB,
+					  CONFIG_LTE_LC_TAU_PRE_WARNING_TIME_MS,
+					  CONFIG_LTE_LC_TAU_PRE_WARNING_THRESHOLD_MS);
 		if (err) {
-			LOG_WRN("%s failed (%d), TAU pre-warning notifications are not enabled",
-				log_strdup(buf_sub), err);
-			LOG_WRN("%s is supported in nRF9160 modem >= v1.3.0", log_strdup(buf_sub));
+			LOG_WRN("Enabling TAU pre-warning notifications failed, error: %d", err);
+			LOG_WRN("TAU pre-warning notifications require nRF9160 modem >= v1.3.0");
 		}
 	}
 
 	if (IS_ENABLED(CONFIG_LTE_LC_MODEM_SLEEP_NOTIFICATIONS)) {
-		snprintk(buf_sub,
-			 sizeof(buf_sub),
-			 AT_XMODEMSLEEP_SUB,
-			 CONFIG_LTE_LC_MODEM_SLEEP_PRE_WARNING_TIME_MS,
-			 CONFIG_LTE_LC_MODEM_SLEEP_NOTIFICATIONS_THRESHOLD_MS);
-
 		/* %XMODEMSLEEP notifications subscribe */
-		err = at_cmd_write(buf_sub, NULL, 0, NULL);
+		err = nrf_modem_at_printf(AT_XMODEMSLEEP_SUB,
+					  CONFIG_LTE_LC_MODEM_SLEEP_PRE_WARNING_TIME_MS,
+					  CONFIG_LTE_LC_MODEM_SLEEP_NOTIFICATIONS_THRESHOLD_MS);
 		if (err) {
-			LOG_WRN("%s failed (%d), modem sleep notifications are not enabled",
-				log_strdup(buf_sub), err);
-			LOG_WRN("%s is supported in nRF9160 modem >= v1.3.0", log_strdup(buf_sub));
+			LOG_WRN("Enabling modem sleep notifications failed, error: %d", err);
+			LOG_WRN("Modem sleep notifications require nRF9160 modem >= v1.3.0");
 		}
 	}
 
 	/* +CSCON notifications */
-	err = at_cmd_write(cscon, NULL, 0, NULL);
+	err = nrf_modem_at_printf(cscon);
 	if (err) {
 		char buf[50];
 
+		/* Ensure null-termination of buffer */
+		buf[sizeof(buf) - 1] = '\0';
+
 		/* AT+CSCON is supported from modem firmware v1.1.0, and will
 		 * not work for older versions. If the command fails, RRC
 		 * mode change notifications will not be received. This is not
 		 * considered a critical error, and the error code is therefore
 		 * not returned, while informative log messageas are printed.
 		 */
-		LOG_WRN("%s failed (%d), RRC notifications are not enabled",
-			cscon, err);
-		LOG_WRN("%s is supported in nRF9160 modem >= v1.1.0", cscon);
+		LOG_WRN("AT+CSCON failed (%d), RRC notifications are not enabled", err);
+		LOG_WRN("AT+CSCON is supported in nRF9160 modem >= v1.1.0");
 
-		err = at_cmd_write("AT+CGMR", buf, sizeof(buf), NULL);
+		err = nrf_modem_at_cmd(buf, sizeof(buf) - 1, "AT+CGMR");
```

Comment:
See comment above, I think there should be no need to decrement one here.

Reply:
Like Tommi said, you must provide the size of the buffer here so simply `sizeof(buf)`.